src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -248,22 +286,51 @@ else if (courseWideContext == null && exerciseId == null && lectureId != null) {
 
     /**
      * Checks course, user and post validity,
-     * retrieves all posts with a certain course-wide context by course id
+     * retrieves and filters posts for a course by its id
      * and ensures that sensitive information is filtered out
      *
-     * @param courseId          id of the course the post belongs to
-     * @param courseWideContext specific course-wide context to filter course get posts for
-     * @return list of posts for a certain course-wide context
+     * @param postContextFilter filter object
+     * @return page of posts that belong to the course
      */
-    public List<Post> getAllPostsByCourseWideContext(Long courseId, CourseWideContext courseWideContext) {
+    public List<Post> getAllCoursePosts(PostContextFilter postContextFilter) {
         final User user = userRepository.getUserWithGroupsAndAuthorities();
 
         // checks
-        final Course course = preCheckUserAndCourse(user, courseId);
+        final Course course = preCheckUserAndCourse(user, postContextFilter.getCourseId());
         authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
 
         // retrieve posts
-        List<Post> coursePosts = postRepository.findPostsForCourseWideContext(courseId, courseWideContext);
+        List<Post> coursePosts;
+        coursePosts = postRepository.findPostsForCourse(postContextFilter.getCourseId(), null, postContextFilter.getFilterToUnresolved(), postContextFilter.getFilterToOwn(),
+                postContextFilter.getFilterToAnsweredOrReacted(), user.getId());
+
+        // protect sample solution, grading instructions, etc.
+        coursePosts.stream().map(Post::getExercise).filter(Objects::nonNull).forEach(Exercise::filterSensitiveInformation);
+
+        return coursePosts;
+    }
+
+    /**
+     * Checks course, user and post validity,
+     * retrieves and filters posts with a certain course-wide context by course id
+     * and ensures that sensitive information is filtered out
+     *
+     * @param postContextFilter filter object
+     * @return page of posts for a certain course-wide context
+     */
+    public List<Post> getAllPostsByCourseWideContext(PostContextFilter postContextFilter) {
+        final User user = userRepository.getUserWithGroupsAndAuthorities();
+
+        // checks
+        final Course course = preCheckUserAndCourse(user, postContextFilter.getCourseId());
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
```

Comment:
Isn't this already done by `preCheckUserAndCourse()`? This seems to be redundant to me.

Reply:
The code was implemented multiple times by a previous developer. I noticed that within "preCheckUserAndCourse", checkHasAtLeastRoleInCourseElseThrow is called with user as parameter. In line 326, it is called without a user. We can remove if it can be confirmed that the behavior wont change.